### DISCUSSION TITLE
怎样部署在内网用ip访问？

### DISCUSSION THREAD

**Discussion by qinxiandiqi** [tryzealot/zealot#1419] [Q&A] [answered]
想要部署在公司内网的一台机器上，用ip来访问，没有域名，应该怎么处理https请求和证书的问题呢？.env文件中的域名字段，我填了机器的ip，证书部分我填了个申请Let's Encrypt的邮箱。部署成功后，https访问会报ERR_SSL_PROTOCOL_ERROR错误，http访问就正常，但是页面和功能跳转会报错。我又换成自签名SSL证书，部署后还是一样的结果。

**Comment by icyleaf** [comment]
看着 #1405

**Reply by qinxiandiqi** [reply]
这个我在系统设置里面设置了，服务器也重启了，但结果还是一样。我又发现浏览器地址栏的提示是“此网站没有证书”，我在.env配置文件夹中配置了自签名证书：
```text
ZEALOT_CERT=/home/ubuntu/docker/zealot-docker/cert/server.crt
ZEALOT_CERT_KEY=/home/ubuntu/docker/zealot-docker/cert/server.key
```
对应路径的文件都是存在的，是我的配置有什么问题吗？

**Reply by icyleaf** [reply, marked as answer]
仅凭你给的信息我无法判断任何内容。请恕我无能为力。我能给到的帮助是：

Let's Encrypt 仅支持域名生成的证书，建议找公司运维配置一个内网域名，否则会真的很麻烦，强烈建议一字不差的阅读完[Docker 部署 Zealot 指南](https://zealot.ews.im/zh-Hans/docs/self-hosted/deployment/docker)并按照里面的步骤操作，如果选择错误可以删除 `docker-compose.yaml` 后重新再来。

**Comment by poapoauu** [comment]
内网环境部署如果使用Let's Encrypt方式 因为是内网机器根本无法获取证书，如果不使用任何证书现在程序提供的请求会被强制改为https 导致需要手动改为http 综上所述 这个项目不适合内网部署

**Reply by icyleaf** [reply]
你这么回答只能显得你很无知，请好好搜索再回答。
